### PR TITLE
Use new CF extension point to avoid needing to parse CM annotations for error reporting

### LINKED ISSUE #140: Error message formatting does not work with Checker Framework master branch
Our logic for adding nicer error message formatting with the `finalizer.invocation.invalid` error type no longer works when using the master branch of the Checker Framework.  We should keep this in mind when updating to the next Checker Framework release.

Unfortunately I don't think we have a regression test checking for the error message formatting.  @kelloggm is there any way to add such a test?

/cc @mernst

### PR DESCRIPTION
**Pull request by kelloggm**
Fixes #140

### CODE REVIEW

**Comment by msridhar** [comment]
Do we match on the exact error message in any of our tests, to make sure this works?

File: object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionVisitor.java [hunk 1]
```diff
@@ -54,4 +60,31 @@ public Void visitMethodInvocation(MethodInvocationTree node, Void p) {
     }
     return super.visitMethodInvocation(node, p);
   }
+
+  @Override
```
Comment (msridhar) [inline object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionVisitor.java:64]: Can we add Javadoc for this method, like we had before for the `reportError()` override?

Comment (kelloggm) [reply object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionVisitor.java:68]: Good point; I put it back.

**Comment by kelloggm** [comment]
> Do we match on the exact error message in any of our tests, to make sure this works?

We do not, but I tested it by hand and the output looks the same. We do test that the error code is `finalizer.invocation.invalid`, though, so the only thing that could go wrong is the error message itself.

**Comment by mernst** [comment]
> It would be great if we could test the actual error message somehow. But that may not be possible with standard CF testing infrastructure at the moment

To do so, you would write a jtreg test.  There are tests in the CF repository like that.

**Comment by msridhar** [comment]
> To do so, you would write a jtreg test. There are tests in the CF repository like that.

Ok.  As far as I can see, this functionality is not exposed in the `framework-test` artifact.  But when we move the checker over to the Checker Framework repository we should add such a test.

**Pull request merged**